src/swap/SwapReviewScreen.tsx

```diff
@@ -228,9 +229,19 @@ export function SwapReviewScreen({ route }: Props) {
 
     // New flow for viem
     if (quote && userInput) {
+      if (quote?.preparedTransactions?.type !== 'possible') {
```

Comment:
could we guarantee this with static typing instead of runtime errors?

Reply:
Yes probably, this screen is going away and we'll improve that.